File: apps/python/lib/chapter11/__main__.py
from lib.chapter11.experiments import N_SPLITS, evaluate_cinema, evaluate_survived
from lib.dataset import data_dir


def main() -> None:
    print(f"Survived（決定木、{N_SPLITS} 分割交差検証の平均）")
    for name, score in evaluate_survived(data_dir() / "Survived.csv").items():
        print(f"  {name}: {score:.4f}")
    print(f"cinema（線形回帰、{N_SPLITS} 分割交差検証の平均）")
    for name, score in evaluate_cinema(data_dir() / "cinema.csv").items():
        print(f"  {name}: {score:.2f}")


if __name__ == "__main__":
    main()

File: apps/node/src/chapter11/main.ts
import { join } from "node:path";
import { dataDir } from "../dataset.ts";
import { N_SPLITS, evaluateCinema, evaluateSurvived } from "./experiments.ts";

export function main(print: (line: string) => void = console.log): void {
  print(`Survived（決定木、${N_SPLITS} 分割交差検証の平均）`);
  const survived = evaluateSurvived(join(dataDir(), "Survived.csv"));
  for (const [name, score] of Object.entries(survived)) {
    print(`  ${name}: ${score.toFixed(4)}`);
  }
  print(`cinema（線形回帰、${N_SPLITS} 分割交差検証の平均）`);
  const cinema = evaluateCinema(join(dataDir(), "cinema.csv"));
  for (const [name, score] of Object.entries(cinema)) {
    print(`  ${name}: ${score.toFixed(2)}`);
  }
}

// node src/chapter11/main.ts で直接実行したときだけ main を呼ぶ
if (import.meta.filename === process.argv[1]) {
  main();
}
